codegenerator/cli/templates/static/codegen/src/IO.res

```diff
@@ -112,22 +114,16 @@ let executeDbFunctionsEntity = (
     ->InMemoryStore.EntityTables.get(entityMod)
     ->InMemoryTable.Entity.rows
 
-  let (entitiesToSet, idsToDelete) = rows->Belt.Array.reduce(([], []), (
-    (accumulatedSets, accumulatedDeletes),
-    row,
-  ) =>
+  let entitiesToSet = []
+  let idsToDelete = []
+
+  rows->Array.forEach(row => {
     switch row {
-    | Updated({latest: {entityUpdateAction: Set(entity)}}) => (
-        Belt.Array.concat(accumulatedSets, [entity]),
-        accumulatedDeletes,
-      )
-    | Updated({latest: {entityUpdateAction: Delete, entityId}}) => (
-        accumulatedSets,
-        Belt.Array.concat(accumulatedDeletes, [entityId]),
-      )
-    | _ => (accumulatedSets, accumulatedDeletes)
+    | Updated({latest: {entityUpdateAction: Set(entity)}}) => entitiesToSet->Array.push(entity)
+    | Updated({latest: {entityUpdateAction: Delete, entityId}}) => idsToDelete->Array.push(entityId)
+    | _ => ()
     }
-  )
```

Comment:
I was playing with the profiler this weekend, and the code happened to create the most memory allocations for the Metri indexer. It is not the biggest memory usage since GC cleans up this correctly, but it affects the performance in a bad way.

Reply:
Interesting!

It seems like lots of people try avoid reduce in general - here I assume it re-allocated each array each iteration?

Reply:
It reallocates the tuple in accumulator + it has concat on every item